### DISCUSSION TITLE
New behavior of making window foreground if launched from shell

### DISCUSSION THREAD

**Discussion by Ulenspiegel** [goldendict/goldendict#1478] [Q&A]
I've been using GoldenDict in a rather niche scenario for a while (a writer's helper), where GD window was kept running, and another application, whenever a new word was under cursor, was initiating a lookup through shell command (GoldenDict.exe "word"), so GD window was serving as a companion real-time thesaurus app showing lookup for current word under the caret as it was typed. It was working smoothly in v1.5.0-RC2-152, because shell command wasn't causing GD window to steal focus from window where the word was being typed, but somewhere on the way to RC2-372 there must've been a change that always brings window to foreground. So, the old use scenario isn't possible anymore. I wonder if it was a deliberate change? Any way to emulate old behavior, or maybe a better way of achieving same goal (e.g. keeping GD as a companion window to automatically show lookups for typed words)?

**Comment by nonwill** [comment]
![Video_20220401110827](https://user-images.githubusercontent.com/46510529/161188145-6f7983c7-6123-4f3f-80ea-3ca493eb5c36.gif)

To satisfy to the habits of all users, how about [adding an option to enable/disable the auto-focus behavior](https://www.autoptr.top/gdocr/GoldenDict-OCR-Changelog/)?
